### PR TITLE
[MI-207] Update oAuth to use Guzzle

### PR DESCRIPTION
**Pull request by joseconsador**
Update oAuth to use Guzzle, update the sample oAuth flow.

/cc @miogalang @jwswj @mmolina

### References
 - Jira link: https://zendesk.atlassian.net/browse/MI-207

### Risks
 - Medium: The client might not make the correct API call.

### CODE REVIEW

**Comment by jwswj** [comment]
What do you think about moving `index.php` and `oauth.php` into a `samples` or `examples` folder?

**Comment by joseconsador** [comment]
@jwswj good suggestion. :haircut:

File: samples/apikey.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?php
```
Comment (jwswj) [inline samples/apikey.php]: We should call this `getTickets.php` rather than `apikey`

File: src/Zendesk/API/Utilities/OAuth.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+     * Requests for an access token.
+     *
+     * @param Client $client
+     * @param string $subDomain
+     * @param array  $params
+     *
+     * @return array
+     * @throws ApiResponseException
+     */
+    public static function getAccessToken(Client $client, $subDomain, array $params)
```
Comment (jwswj) [inline src/Zendesk/API/Utilities/OAuth.php]: why capital D in subdomain?

File: tests/Zendesk/API/UnitTests/Utilities/OAuthTest.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+class OAuthTest extends BasicTest
+{
+    /**
+     * Tests if the OAuth::getAuthUrl function returns a correct URL.
+     */
+    public function testAuthUrl()
+    {
+        $params = [
+            'client_id'    => 'test',
+            'state'        => 'test',
```
Comment (jwswj) [inline tests/Zendesk/API/UnitTests/Utilities/OAuthTest.php]: test and test should be different values

File: tests/Zendesk/API/UnitTests/Utilities/OAuthTest.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        ];
+
+        $expected = 'https://test.zendesk.com/oauth/authorizations/new?response_type=code&client_id=test&state=test&scope=read+write';
```
Comment (jwswj) [inline tests/Zendesk/API/UnitTests/Utilities/OAuthTest.php]: subdomain should also be something different than test

**Comment by jwswj** [comment]
with those small changes I'm a :+1:

**Comment by miogalang** [comment]
:+1:

**Pull request merged**